src/components/controllers/slotted-controller.ts

```diff
@@ -1,12 +1,16 @@
-import { ReactiveControllerHost, ReactiveController } from 'lit';
+import {
+  ReactiveControllerHost,
+  ReactiveController,
+  ReactiveElement,
+} from 'lit';
 
 export class SlottedController implements ReactiveController {
   host: ReactiveControllerHost;
-  hostEl: HTMLElement;
+  hostEl: ReactiveElement;
 
   constructor(host: ReactiveControllerHost) {
     (this.host = host).addController(this);
-    this.hostEl = this.host as unknown as HTMLElement;
+    this.hostEl = this.host as unknown as ReactiveElement;
   }
 
   hostConnected() {
```

Comment:
Can you listen to `slotchange` on a web component? I thought this event only happens on the slotted elements themselves. I could be wrong.

https://developer.mozilla.org/en-US/docs/Web/API/HTMLSlotElement/slotchange_event

If you can't, this code just doesn't ever fire.

Reply:
@damontgomery You are correct, you have to attach the event to the slotted element itself. I tried to do this on a component for SMC and I can confirm it does not work.

Reply:
@mabry1985 can you give me more details of what works or doesn't work?
(I don't think I added this part of the code, so I am not familiar with it)

Reply:
`slotchange` events fire on `slot` elements and don't bubble. So, you can add `@slotchange=...` But the web component (parent) doesn't get those events.

So, if you want to respond to slot changes, you need to watch all the slots. Or, you can add a mutation observer on the light DOM element.

```
  observer: MutationObserver;

  connectedCallback() {
    super.connectedCallback();

    this.observer = new MutationObserver(() => {
      this.requestUpdate();
    });
    this.observer.observe(this, { childList: true });
  }

  disconnectedCallback() {
    this.observer.disconnect();
  }
```

Reply:
Anyone have time to dig into this a little more? This sounds interesting.